### PR TITLE
Add DEVELOPMENT.md

### PR DESCRIPTION
**Pull request by wlynch**
- Creates DEVELOPMENT.md with details on required tools and how to use
  e2e test scripts to bootstrap local development.
- Revamps README.md to include data model overview and provide contact
  links.
- Renames e2e test scripts to make it clear the order they should be ran
  in.

Note: Some relative links/images are broken because GitHub is linking them to the main branch. These should work once submitted. 🤞

### CODE REVIEW

File: test/e2e/README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ this environment.
 | KIND_CLUSTER_NAME    | KIND cluster name to use | tekton-results        |
 | KIND_IMAGE           | KIND node image to use   | kindest/node:v1.17.11 |
 
-### `install.sh`
+### `01-install.sh`
```
Comment (XinruZhang) [inline test/e2e/README.md:33]: the same for `test.sh` on line 49

**Review by XinruZhang — COMMENTED** [review]
Looks good!

File: test/e2e/README.md (hunk 1 above)
Comment (wlynch) [reply test/e2e/README.md:33]: Done.

**Comment by bobcatfish** [comment]
nice!!! :D

**Pull request merged**